lib/srv/desktop/rdp/rdpclient/src/rdpdr/consts.rs

```diff
@@ -88,7 +88,7 @@ pub enum MajorFunction {
     IRP_MJ_LOCK_CONTROL = 0x00000011,
 }
 
-#[derive(Debug, FromPrimitive, ToPrimitive)]
+#[derive(Debug, FromPrimitive, ToPrimitive, Clone)]
```

Comment:
Since it's just numbers, derive `Copy` and remove the clone calls that needed this.

Reply:
Can you explain more? Not doubting you, I just don't get the distinction.

Reply:
If I change this to Copy, the compiler complains at me:
```
the trait bound `MinorFunction: Clone` is not satisfied
the trait `Clone` is not implemented for `MinorFunction`
```

Reply:
You can't just use `Copy` trait alone. `Clone` is a supertrait of `Copy`, so everything which is `Copy` must also implement `Clone`. You can check for more information in the [rust docs](https://doc.rust-lang.org/std/marker/trait.Copy.html#whats-the-difference-between-copy-and-clone). What @xacrimon meant is that you have to derive `Copy` and `Clone` traits together, and then remove any clone calls from the code that needed that `Clone` trait.

Reply:
Yep ^^